### PR TITLE
Replace Throughput Bugfix

### PR DESCRIPTION
**Pull request by andrewmathew1**
# Description

Un-indented two if statements from being under the max throughput if statement logic. 
This is to address a bug where users could only manually replace throughput by passing an int. 
Now users can pass the following successfully:
```python
new_throughput = ThroughputProperties(offer_throughput=2500)
update_throughput = collection.replace_throughput(new_throughput)
```

### CODE REVIEW

**Review by simorenoh — COMMENTED** [review]
Let's please add the async test as well, and let's add some tests for negative scenarios if we don't have them (what if I try to replace using ThroughputProperties that has all the configs?)

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 1]
```diff
@@ -795,10 +795,10 @@ def _replace_throughput(
             increment_percent = throughput.auto_scale_increment_percent
             if max_throughput is not None:
                 new_throughput_properties['content']['offerAutopilotSettings']['maxThroughput'] = max_throughput
-                if increment_percent:
-                    new_throughput_properties['content']['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy']['incrementPercent'] = increment_percent  # pylint: disable=line-too-long
-                if throughput.offer_throughput:
-                    new_throughput_properties["content"]["offerThroughput"] = throughput.offer_throughput
+            if increment_percent:
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py]: I think increment percent should only be set if max throughput is set. If you look at offer.py it explains which throughput properties should be set together. We should also add a test for this if there isn't one already.

**Review by tvaron3 — CHANGES_REQUESTED** [review]
LGTM besides small comment

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py (hunk 1 above)
Comment (andrewmathew1) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_base.py]: Great point! I went ahead and moved increment percent back inside that if statement. I looked and there are tests for setting throughput with auto scale max throughput and auto scale increment percent in the container_management.py and container_management_async.py files.

**Review by tvaron3 — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/tests/test_crud_container.py [hunk 2]
```diff
@@ -954,7 +954,35 @@ def test_index_progress_headers(self):
...
+        retrieve_throughput = collection.get_throughput()
+        assert getattr(retrieve_throughput, "offer_throughput") == getattr(new_throughput, "offer_throughput")
+
+    def test_negative_replace_throughput_with_all_configs_set(self):
+        created_db = self.databaseForTest
+        collection = created_db.get_container_client(self.configs.TEST_MULTI_PARTITION_CONTAINER_ID)
+
+        new_throughput = ThroughputProperties(offer_throughput=2500, auto_scale_max_throughput=4000, auto_scale_increment_percent=5)
+        try:
+            collection.replace_throughput(new_throughput)
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos/tests/test_crud_container.py:984]: Can you use pytest.raises to make show that an error appears when customer tries to set with all the options? Right now this test would pass even if there wasn't an error.
